src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/halapi.cxx

```diff
@@ -537,6 +390,19 @@ int xclPollCompletion(xclDeviceHandle handle, int min_compl, int max_compl, xclR
   return drv ? drv->xclPollCompletion(min_compl, max_compl, comps, actual, timeout) : -ENODEV;
 }
 
+size_t xclDebugReadIPStatus(xclDeviceHandle handle, xclDebugReadType type, void* debugResults)
+{
+  return 0;
+}
+
+double xclGetDeviceClockFreqMHz(xclDeviceHandle handle)
+{
+  xclhwemhal2::HwEmShim *drv = xclhwemhal2::HwEmShim::handleCheck(handle);
```

Comment:
auto?

Reply:
Is auto required here? The required type is already used here.

Reply:
It's not.